instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/datasource/OpenTelemetryDataSourceBuilder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jdbc.datasource;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcInstrumenterFactory;
+import javax.sql.DataSource;
+
+public class OpenTelemetryDataSourceBuilder {
+
+  private final DataSource delegate;
+
+  private final OpenTelemetry openTelemetry;
+
+  private boolean dataSourceInstrumenterEnabled = true;
+
+  private boolean statementInstrumenterEnabled = true;
+
+  private boolean statementSanitizationEnabled = true;
+
+  OpenTelemetryDataSourceBuilder(DataSource delegate, OpenTelemetry openTelemetry) {
+    this.delegate = delegate;
+    this.openTelemetry = openTelemetry;
+  }
+
+  @CanIgnoreReturnValue
+  public OpenTelemetryDataSourceBuilder enableDataSourceInstrumenter(boolean enabled) {
```

Comment:
Perhaps something like this would be more "fluent", "natural":
```java
 @CanIgnoreReturnValue
  public OpenTelemetryDataSourceBuilder disableDataSourceInstrumenter() {
    this.dataSourceInstrumenterEnabled = false;
    return this;
  }
```

The instrumenter is enabled by default.

Same comment for the following methods.

Reply:
I opted to allow a boolean to be passed through as I thought this might be more natural for situations where you are enabling/disabling the instumenters via feature flag. You can just pass in your variable and have less verbose code. Happy to be challenged here.

Alternatively we can rename the methods to something like `setDataSourceInstrumenterEnabled(true)`.

Reply:
I've updated this PR so that this class and method no longer exist.